### PR TITLE
feat: add chart component

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds chart component (no data integration and won't be displayed yet)

Bildschirmfoto 2024-02-27 um 14 46 25
Bildschirmfoto 2024-02-27 um 14 47 28

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/app/products/page.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      <Flex maxWidth={'500px'} w={'100%'} h={'100%'}>
+        <Swap
+          isBuying={isMinting}
+          inputToken={inputToken}
+          outputToken={outputToken}
+        />
+      </Flex>
+    </Flex>
+  )
+}
```
Comment (janndriessen) [inline src/app/products/page.tsx:29]: Just for testing, will be hidden later.

File: src/components/chart/components/line-chart.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+import { localPoint } from '@visx/event'
+import { LinearGradient } from '@visx/gradient'
+import { max, extent, bisector } from '@visx/vendor/d3-array'
+import { timeFormat } from '@visx/vendor/d3-time-format'
+
+import { colors } from '@/lib/styles/colors'
+
+type TooltipData = AppleStock
+
+export const accentColor = '#05ACAF'
```
Comment (0xonramp) [inline src/components/chart/components/line-chart.tsx:20]: should we add to the color palette, or not used elsewhere?

File: src/components/chart/components/range-selection.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  onClick: () => void
+}
+
+function RangeSelectionItem({
+  label,
+  onClick,
+  isSelected,
+}: RangeSelectionItemProps) {
+  const color = isSelected ? '#1C2929' : colors.icGray5
+  const backgroundColor = isSelected ? '#F2F2F2' : 'none'
```
Comment (0xonramp) [inline src/components/chart/components/range-selection.tsx:19]: similar questions here RE: palette

Comment (janndriessen) [reply src/components/chart/components/range-selection.tsx]: Updated colors according to what's available in interfaces. `#F2F2F2` wasn't available so leaving it as custom for now.

**Pull request merged**